_rmd/2017-11-27-spreadsheet_classification.Rmd

```diff
@@ -0,0 +1,297 @@
+
+However, these demonstrative examples on model data sets can inform our decision making for feature extraction by standing on the shoulders of giants:  
+
+* This [ENRON paper](https://figshare.com/articles/Enron_s_Spreadsheets_and_Related_Emails_A_Dataset_and_Analysis/1222882) also identifies some useful proxy measures of a spreadsheets quality by categorising the "smells" in a spreadsheets formulae (e.g. unique formulas  that have a claculation chain of five or longer).  
+* This could be useful for an organisation attempting to audit the quality of its spreadsheets and flagging spreadsheets for quality assurance.  
+
+## The data
+
+The data is pretty big (almost 1 Gb zipped) so we don't store it in our repo. Instead download it from origin at [figshare]([Enron spreadsheets](https://figshare.com/articles/Enron_Spreadsheets_and_Emails/1221767)) (Hermanns, 2015). Unfortunately this data is in .xlsx format.  
+
+Some interesting points about this Enron spreadsheet data, especially if it is representative of other organisations (albeit it is somewhat dated now):  
+
+* 24% of Enron spreadsheets with formulas contain an Excel error. It's also noted that formal testing (software development) is rare in spreadsheets in general, discussed in a [recent paper](http://swerl.tudelft.nl/twiki/pub/Main/TechnicalReports/TUD-SERG-2017-002.pdf), the very thing required to protect against these errors.  
+* There is remarkably little diversity in the functions used in spreadsheets: we observe that there is a core set of 15 spreadsheet functions which is used in 76% of spreadsheets. Thus, these could be readily replaced by a software package containing this core set used across the organisation and developed using DevOps best practice. This  supports the case for developing [reproducible analytical pipelines](https://ukgovdatascience.github.io/rap_companion/) in an organisation.      
+* Spreadsheet use within companies in common, with 100
+spreadsheets emailed around per day! Version control is essential for collaboration, yet looks like Enron didn't use it...      
+
+## Reading the data
+
+We copy and modify [Duncan's code](https://github.com/nacnudus/ClassifySpreadsheets/tree/master/analysis) to read in the spreadsheets. It's great practice to work with another data scientist on a project to help you learn new tips and tricks (for example, I hadn't seen the `here` package before).
+
+```{r}
+# Googled it, it's not on CRAN yet
```

Comment:
[here](https://cran.r-project.org/package=here)
[piton](https://cran.r-project.org/package=piton)
[tidyxl](https://cran.r-project.org/package=tidyxl)

Reply:
@nacnudus, I've just added you to the org, so check your email!